ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java

```diff
@@ -250,14 +261,17 @@ public static void main(String[] args) {
               clientArgs.serviceId, clientArgs.targetAccountName, clientArgs.targetContainerName,
               clientArgs.customHeaders);
       // attach shutdown handler to catch control-c
-      Runtime.getRuntime().addShutdownHook(new Thread() {
-        public void run() {
-          logger.info("Received shutdown signal. Requesting NettyPerfClient shutdown");
-          nettyPerfClient.shutdown();
-        }
-      });
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+        logger.info("Received shutdown signal. Requesting NettyPerfClient shutdown");
+        nettyPerfClient.shutdown();
+      }));
       nettyPerfClient.start();
-      nettyPerfClient.awaitShutdown();
+      if (clientArgs.testTime != null) {
+        ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+        scheduler.schedule(nettyPerfClient::shutdown, clientArgs.testTime, TimeUnit.SECONDS);
+        nettyPerfClient.awaitShutdown();
+        scheduler.shutdownNow();
```

Comment:
`scheduler.awaitTermination()`?

Reply:
also, I think the part including and beyond `nettyPerfClient.awaitShutdown()` should be outside the `if`?

Reply:
will move to outside the if. Also, I thought that we had to use `shutdownNow` to kill future scheduled tasks, but that's actually not true. I switched to using `Utils.shutdownExecutorService` for consistency.